### PR TITLE
Added new Email type

### PR DESCRIPTION
**Pull request by romaninsh**
Re-implementation of https://github.com/atk4/data/pull/443

Adds new field type `Email` with various options for validation and safe storage of emails:

``` php
$user->addField('email', ['Email']);
$user->addField('email_mx_check', ['Email', 'dns_check'=>true]);
$user->addField('email_with_name', ['Email', 'include_names'=>true]);
$user->addField('emails', ['Email', 'allow_multiple'=>true]);
```

Various options can also be combined. Also will clean up spaces. @acicovic - what do you think ?

### CODE REVIEW

**Comment by acicovic** [comment]
I think this is great and it's an excellent demonstration on how to create a simple custom field.

File: src/Field/Email.php [hunk 1]
```diff
@@ -25,34 +35,66 @@ class Email extends Field
...
+        $emails = array_map(function ($email) {
             $email = trim($email);
 
             if ($this->include_names) {
                 $email = preg_replace('/^[^<]*<([^>]*)>/', '\1', $email);
             }
 
-            if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
-                throw new Exception(['Email format is invalid', 'email'=>$email]);
+            // should actually run only domain trough idn_to_ascii(), but for validation purpose this way it's fine too
```
Comment (romaninsh) [inline src/Field/Email.php]: test-case for this?

Comment (romaninsh) [reply src/Field/Email.php]: oh i see it!

Comment (DarkSide666) [reply src/Field/Email.php]: this comment line (69) should actually be removed now i guess :)

**Pull request merged**